initializr-generator/src/main/java/io/spring/initializr/metadata/InitializrMetadata.java

```diff
@@ -282,19 +282,28 @@ public String getContent() {
 	}
 
 	private static class PackageCapability extends TextCapability {
-		private final TextCapability nameCapability;
+		private final TextCapability groupId;
+		private final TextCapability artifactId;
 
-		PackageCapability(TextCapability nameCapability) {
+		PackageCapability(TextCapability groupId, TextCapability artifactId) {
 			super("packageName", "Package Name", "root package");
-			this.nameCapability = nameCapability;
+			this.groupId = groupId;
+			this.artifactId = artifactId;
 		}
 
 		@Override
 		public String getContent() {
 			String value = super.getContent();
-			return value != null ? value
-					: (nameCapability.getContent() != null
-							? nameCapability.getContent().replace("-", ".") : null);
+			if (value != null) {
+				return value;
+			}
+			else if(this.groupId.getContent() != null && this.artifactId.getContent() != null) {
+				return this.groupId.getContent()
+						.concat(".")
+						.concat(this.artifactId.getContent())
+						.replaceAll("-", "");
```

Comment:
We're not changing the `groupId` value, but just limiting the depth of the `package` value.
Applying the same rule for the package would yield `com.example.user.management` instead of `com.example.usermanagement`.

I don't have a strong opinion on this.